test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJcmd.java

```diff
@@ -0,0 +1,183 @@
+			log(EXPECTED_STRING_FOUND);
+		}
+	}
+
+	@Test
+	public void testListVms() throws IOException {
+		String myId = getVmId();
+		List<String> args = new ArrayList<>();
+		List<String> jpsOutput = runCommandAndLogOutput(args);
+		Optional<String> searchResult = StringUtilities.searchSubstring(myId, jpsOutput);
+		String errorMessage = "My VMID missing from VM list"; //$NON-NLS-1$
+		assertTrue(searchResult.isPresent(), errorMessage);
+		args.add("-l"); //$NON-NLS-1$
+		jpsOutput = runCommandAndLogOutput(args);
+		searchResult = StringUtilities.searchSubstring(myId, jpsOutput);
+		assertTrue(searchResult.isPresent(), errorMessage);
+		log(EXPECTED_STRING_FOUND);
+	}
+
+	@Test
+	public void testCommandNoOptions() throws IOException {
+		for (String command : JCMD_COMMANDS) {
+			List<String> args = new ArrayList<>();
+			args.add(getVmId());
+			args.add(command);
+			List<String> jpsOutput = runCommandAndLogOutput(args);
+			String expectedString = commandExpectedOutputs.getOrDefault(command, "Test error: expected output not defined"); //$NON-NLS-1$
+			log("Expected string: " + expectedString); //$NON-NLS-1$
+			Optional<String> searchResult = StringUtilities.searchSubstring(expectedString, jpsOutput);
+			assertTrue(searchResult.isPresent(), "Expected string not found"); //$NON-NLS-1$
+			log(EXPECTED_STRING_FOUND);
+		}
+	}
+
+	@Test
+	public void testThreadPrint() throws IOException {
+		String LockedSyncsOption = "-l"; //$NON-NLS-1$
+		@SuppressWarnings("nls")
+		String[] options = {"", LockedSyncsOption, LockedSyncsOption + "=true", LockedSyncsOption + "=false"};
+		for (String option : options) {
+			List<String> args = new ArrayList<>();
+			args.add(getVmId());
+			args.add(THREAD_PRINT);
+			if (!option.isEmpty()) {
+				args.add(option);
+			}
+			boolean addSynchronizers = !option.endsWith("false") && !option.isEmpty(); //$NON-NLS-1$
+			List<String> jpsOutput = runCommandAndLogOutput(args);
+			String expectedString = "Locked ownable synchronizers"; //$NON-NLS-1$
+			log("Expected string: " + expectedString); //$NON-NLS-1$
+			Optional<String> searchResult = StringUtilities.searchSubstring(expectedString, jpsOutput);
+			assertEquals(searchResult.isPresent(), addSynchronizers, "Output contains locked synchronizer information"); //$NON-NLS-1$
+			log(EXPECTED_STRING_FOUND);
+		}
+	}
+
+	@Test
+	public void testClassHistogramAll() throws IOException {
+		List<String> args = new ArrayList<>();
+		args.add(getVmId());
+		args.add(GC_CLASS_HISTOGRAM);
+		args.add("-all"); //$NON-NLS-1$
+		List<String> jpsOutput = runCommandAndLogOutput(args);
+		String expectedString = commandExpectedOutputs.getOrDefault(GC_CLASS_HISTOGRAM, "Test error: expected output not defined"); //$NON-NLS-1$
+		log("Expected string: " + expectedString); //$NON-NLS-1$
+		Optional<String> searchResult = StringUtilities.searchSubstring(expectedString, jpsOutput);
+		assertTrue(searchResult.isPresent(), "Expected string not found"); //$NON-NLS-1$
```

Comment:
The assertion message would be more helpful if it mentioned the string that we expected to find.